reverseforwardrecord/get_dns_record.py

```diff
@@ -0,0 +1,255 @@
+            dns_found.append(line.split('\t')[position_dns])
+    return ips_found, dns_found
+
+
+def is_reachable_ip(ip: str) -> bool:
+    """
+    used to ping an ip and see whether it is reachable or not
+    :param ip: converts ip to string
+    :return: returns boolean result
+    """
+    try:
+        result = subprocess.run(["ping", "-c", "1", ip], capture_output=True)
+        return result.returncode == 0
+    except Exception as e:
+        print(f"Error pinging IP {ip}: {e}")
+        return False
+
+
+def is_reachable_dns(dns_name: str) -> bool:
+    """
+    used to resolve a dns and see whether it is reachable or not
+    :param dns_name: converts dns to string
+    :return: returns boolean result
+    """
+    try:
+        socket.gethostbyname(dns_name)
+        return True
+    except socket.error as e:
+        print(f"Error resolving DNS {dns_name}: {e}")
+        return False
+
+
+def check_reachability(ips_found: List[str], dns_found: List[str]) \
+        -> Tuple[List[str], List[str], List[str], List[str]]:
+    """
+    checks whether the ips and dns names are reachable, using two previous functions and then
+    appending the results of those to the lists which are after displayed to the command line
+    :params ips_found: ips in a list to be tested
+    :params dns_found: dns names in a list to be tested
+    :params tuple_lists: lists that cant be changed
+    :return: all the ips found and dns names found that into 2 sections being reachable meaning
+    working ips or dns or unreachable which are out of use ips or dns that cannot be reached
+    """
+    reachable_ips = []
+    unreachable_ips = []
+    reachable_dns = []
+    unreachable_dns = []
+
+    for ip in ips_found:
+        if is_reachable_ip(ip):
+            reachable_ips.append(ip)
+        else:
+            unreachable_ips.append(ip)
+
+    for dns_name in dns_found:
+        if is_reachable_dns(dns_name):
+            reachable_dns.append(dns_name)
+        else:
+            unreachable_dns.append(dns_name)
+
+    return reachable_ips, unreachable_ips, reachable_dns, unreachable_dns
+
+
+def get_dns_record(
+        netbox_filepath: str,
+        fqdn_column_name: str = "FQDN",
+        idrac_ip_column_name: str = "IDRAC IP",
+        reverse_order: bool = False,
+        output_filepath: str = "output.txt", ):
+    """
+    this function will create a file for the output for dns records
+    also testing each of the ips and dns to see whether they are usable or not printing after
+    to display the result
+    :param netbox_filepath: path for the netbox info
+    :param fqdn_column: full qualified domain name column in netbox file
+    :param idrac_ip_column: idrac ip column in netbox file
+    :param reverse_order: use reverse or forward order / if true use forward if false use reverse
+    :param output_filepath: file path for output
+    :return: None
+    """
+    create_file(output_filepath)
```

Comment:
you don't need this function - see above
```suggestion
    open(out_filepath, 'w').close()
```